src/Vgrep/Environment/Config.hs

```diff
@@ -56,24 +61,71 @@ makeLenses ''Colors
+    { _colors  = fromColorsMonoid _mcolors
+    , _tabstop = fromFirst (_tabstop defaultConfig) _mtabstop
+    , _editor  = fromFirst (_editor  defaultConfig) _meditor }
+
+-- | Convert a 'ColorsMonoid' to a 'Colors' config.
+fromColorsMonoid :: ColorsMonoid -> Colors
+fromColorsMonoid ColorsMonoid{..} = Colors
+    { _lineNumbers   = fromFirst (_lineNumbers   defaultColors) _mlineNumbers
+    , _lineNumbersHl = fromFirst (_lineNumbersHl defaultColors) _mlineNumbersHl
+    , _normal        = fromFirst (_normal        defaultColors) _mnormal
+    , _normalHl      = fromFirst (_normalHl      defaultColors) _mnormalHl
+    , _fileHeaders   = fromFirst (_fileHeaders   defaultColors) _mfileHeaders
+    , _selected      = fromFirst (_selected      defaultColors) _mselected }
+
+fromFirst :: a -> First a -> a
+fromFirst a = fromMaybe a . getFirst
+
+
+--------------------------------------------------------------------------
 -- * Default Config
 --------------------------------------------------------------------------
 
 defaultConfig :: Config
 defaultConfig = Config
-    { _colors = Colors
-        { _lineNumbers   = defAttr `withForeColor` blue
-        , _lineNumbersHl = defAttr `withForeColor` blue
-                                   `withStyle` bold
-        , _normal        = defAttr
-        , _normalHl      = defAttr `withStyle` bold
-        , _fileHeaders   = defAttr `withBackColor` green
-        , _selected      = defAttr `withStyle` standout }
+    { _colors = defaultColors
     , _tabstop = 8
     , _editor = "vi" }
 
-withConfiguredEditor :: Config -> IO Config
-withConfiguredEditor config = do
-    let defaultEditor = view editor config
-    configuredEditor <- lookupEnv "EDITOR"
-    pure config { _editor = fromMaybe defaultEditor configuredEditor }
+defaultColors :: Colors
+defaultColors = Colors
+    { _lineNumbers   = defAttr `withForeColor` blue
+    , _lineNumbersHl = defAttr `withForeColor` blue
+                               `withStyle` bold
+    , _normal        = defAttr
+    , _normalHl      = defAttr `withStyle` bold
+    , _fileHeaders   = defAttr `withBackColor` green
+    , _selected      = defAttr `withStyle` standout }
+
+
+--------------------------------------------------------------------------
+-- * Config Loader
+--------------------------------------------------------------------------
+
+-- | Gathers 'ConfigMonoid's from various sources and builds a 'Config'
+-- based on the 'defaultConfig':
+--
+-- * Config from environment variables
+-- * The configuration specified in the config file
+-- * External config, e.g. from command line
+--
+-- where the latter ones override the earlier ones.
+loadConfig
+    :: MonadIO io
+    => ConfigMonoid -- ^ External config from command line
+    -> io Config
+loadConfig configFromArgs = do
+    configs <- sequence
+        [ pure configFromArgs
+        , configFromFile
+        , editorConfigFromEnv ]
+    pure (fromConfigMonoid (mconcat configs))
```

Comment:
`x <- m; pure (nested (deeply x))` can often be written nicer using sub-`do`:

```haskell
configs <- do
    x <- sequence [...]
    pure (mconcat configs)
pure (fromConfigMonoid configs)
```

Maybe this is not necessary here, but in general I think the `raw <- x; unRaw raw` pattern should be avoided (by never having `raw` data in scope longer than needed, or passed into conveyor belt function chains).